### ISSUE TITLE
testSamConversion fails on ubuntu 14.04 and ubuntu 15.05

### ISSUE THREAD

**Issue by IanRogers-LShift** [ga4gh/server#783]
Running nosetests on latest server develop branch (commit  58b228bd8de761d110de897e50e36447553ba748) on ubuntu 14.04 fails with this error:

FAIL: testSamConversion (tests.unit.test_converters.TestSamConverter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ian/workspace/ga4gh/ga4gh/server/tests/unit/test_converters.py", line 75, in testSamConversion
    self.verifyFullConversion(readGroupSet, readGroup, reference)
  File "/home/ian/workspace/ga4gh/ga4gh/server/tests/unit/test_converters.py", line 66, in verifyFullConversion
    self.verifySamRecordsEqual(sourceReads, convertedReads)
  File "/home/ian/workspace/ga4gh/ga4gh/server/tests/unit/test_converters.py", line 35, in verifySamRecordsEqual
    self.assertEqual(source.flag, converted.flag)
AssertionError: 73 != 65

Also confirmed by someone else to fail in ubuntu 15.04

Environment details (created by following http://ga4gh-reference-implementation.readthedocs.org/en/stable/demo.html )  :

pysam-0.8.3-py2.7 

$ uname -a
Linux mrchatterbox 3.13.0-66-generic #108-Ubuntu SMP Wed Oct 7 15:20:27 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
$ which python
/home/ian/workspace/ga4gh/ga4gh/server/ga4gh-env/bin/python
$ python --version
Python 2.7.6

**Comment by jeromekelleher** [comment]
Thanks @IanRogers-LShift --- also confirmed on a fresh Ubuntu 15.10 installation.

**Comment by dcolligan** [comment]
Could not reproduce on Ubuntu 15.4:
```
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ uname -a
Linux dannyc-VirtualBox 3.19.0-15-generic #15-Ubuntu SMP Thu Apr 16 23:32:37 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ which python
/home/dannyc/ga4gh/cur/bin/python
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ python --version
Python 2.7.9
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 15.04
Release:	15.04
Codename:	vivid

**Comment by dcolligan** [comment]
Could not reproduce on Ubuntu 15.10:
```
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ uname -a
Linux dannyc-VirtualBox 4.2.0-16-generic #19-Ubuntu SMP Thu Oct 8 15:35:06 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ which python
/home/dannyc/ga4gh/cur/bin/python
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ python --version
Python 2.7.10
(cur)dannyc@dannyc-VirtualBox:~/ga4gh/server$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 15.10
Release:	15.10
Codename:	wily
```
The only difference I am using from the install instructions is installing `libssl-dev` as well as using a virtualenv instead of the `--user` flag.

@jeromekelleher do you see any differences from your installation where you could reproduce the issue?

**Comment by jeromekelleher** [comment]
This is weird... Looks like an identical Ubuntu vm. What happens when you install as --user?

**Comment by dcolligan** [comment]
Installed as user, no problems with that test.  However, another test did fail:
```
======================================================================
ERROR: test suite for <class 'tests.end_to_end.test_oidc.TestOidc'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dannyc/.local/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/home/dannyc/.local/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/home/dannyc/.local/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/home/dannyc/.local/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/dannyc/ga4gh/server/tests/end_to_end/server_test.py", line 70, in setUpClass
    cls.otherSetup()
  File "/home/dannyc/ga4gh/server/tests/end_to_end/test_oidc.py", line 58, in otherSetup
    cls.opServer.start()
  File "/home/dannyc/ga4gh/server/tests/end_to_end/server.py", line 243, in start
    stdout=tempFile, stderr=subprocess.STDOUT)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command '[u'./setupenv.sh']' returned non-zero exit status 2
```
This was fixed after an install of the ubuntu package `virtualenv` ... see #788

**Comment by jeromekelleher** [comment]
This is bizarre... are you using a fresh VM? Could you be using an older compiled version of pysam or something, whereas we've made new ones (grasping at straws)?

**Comment by dcolligan** [comment]
No, I created a brand spanking new VM and followed the instructions on our development doc page to install.

**Comment by jeromekelleher** [comment]
OK, maybe its a stochastic effect? How about if you run this test in a bash loop for a while?

**Comment by dcolligan** [comment]
Tried that... no dice.  How about you take this one since I can't reproduce?

**Comment by jeromekelleher** [comment]
Crud. OK, assigning this to me.

**Comment by jeromekelleher** [comment]
OK, I can see why this isn't happening sometimes. We're only choosing one readgroup and the order these occur in is file system dependent. If we change to this code in testSamConversion the bug happens reliably. Looks like it is some problem with the flags conversion.

```python
        for dataset in self._backend.getDatasets():
            for readGroupSet in dataset.getReadGroupSets():
                referenceSet = readGroupSet.getReferenceSet()
                for reference in referenceSet.getReferences():
                    for readGroup in readGroupSet.getReadGroups():
                        self.verifyFullConversion(readGroupSet, readGroup, reference)
```

Over to you @dcolligan!

**Comment by dcolligan** [comment]
Fixed in #789
